### PR TITLE
rfb: add unittests for nom7 parsers

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/5741) ticket:

Describe changes:
- Add unit tests for rfb protocol

### CODE REVIEW

**Comment by hsadia538** [comment]
Currently, the unit tests are written for the following parse functions:
Screenshot 2022-12-09 at 11 33 24 PM

More parse functions can be incorporated but some of them would be testing the same value e.g `parse_security_type_selection ,  parse_server_security_type , parse_supported_security_types ` are quite similar as all of them return the `security type `selected or asked (the difference is only in context).

The packet buffers I have used are actually coming from an rfb pcap that I found in suricata verify. Should I add the reference somewhere in the file?

**Apart from this, the main intention of this PR is to gather feedback on the approach and whether should I try doing the same for other tests that I will write, etc (including formatting, readability, and other factors that matter)**

File: rust/src/rfb/parser.rs [hunk 1]
```diff
@@ -110,46 +110,63 @@ pub struct ServerInit {
     pub height: u16,
     pub pixel_format: PixelFormat,
     pub name_length: u32,
-    pub name: Vec<u8>
+    pub name: Vec<u8>,
 }
 
+//done
```
Comment (victorjulien) [inline rust/src/rfb/parser.rs:116]: what is this comment?

File: rust/src/rfb/parser.rs [hunk 2]
```diff
@@ -36,7 +36,7 @@ pub enum RFBGlobalState {
     TSVncResponse,
     TCSecurityResult,
     TSClientInit,
-    Message
+    Message,
```
Comment (victorjulien) [inline rust/src/rfb/parser.rs:39]: please split formatting changes from the test additions into separate commits

**Review by victorjulien — CHANGES_REQUESTED** [review]

File: rust/src/rfb/parser.rs (hunk 2 above)
Comment (hsadia538) [reply rust/src/rfb/parser.rs:39]: Ok.Will address in next PR

File: rust/src/rfb/parser.rs (hunk 1 above)
Comment (hsadia538) [reply rust/src/rfb/parser.rs:116]: Did not mean to commit these comments. Will remove them now( I added this for parsers whose unit tests are there)

**Comment by hsadia538** [comment]
Followed by #8290

**Pull request closed without merging**